src/controllers/dynakube/version/image_hash.go

```diff
@@ -45,8 +46,17 @@ func GetImageVersion( //nolint:revive // argument-limit
 	var err error
 	var proxy string
 
-	keychain := dockerkeychain.NewDockerKeychain(registryAuthPath, afero.NewOsFs())
 	transport := http.DefaultTransport.(*http.Transport).Clone()
+	keychain := dockerkeychain.NewDockerKeychain()
+	err = keychain.LoadDockerConfigFromSecret(ctx, apiReader, v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dynakube.PullSecret(),
```

Comment:
you could do some refactoring to make you life easier:
`dynakube.PullSecret() -> dynakube.PullSecretName()`

then create a `dynakube.PullSecret()`(with some other name that tells the reader that its an incomplete secret as it does not have the data part of it) that actually returns a secret, so we don't do this all the time.
- or you can make the `dynakube.PullSecret()` to return the actual pull secret (make the call and all), similar how the `dynakube.Proxy()` will make a call to get the proxy

Reply:
Will create a follow up ticket to tackle this

Reply:
Or not, moved the `PullSecret` function to `PullSecretName` and created a new function called `PullSecretWithoutData`